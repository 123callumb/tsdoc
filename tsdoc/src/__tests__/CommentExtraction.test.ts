import { TSDocParser, DocComment } from '../index';

test('Comment extraction', () => {

  const parser: TSDocParser = new TSDocParser();

  const input: string = `
    /**
     * This is an example doc comment.
     * @remarks
     * Here are some remarks
     *
     * @public
     */
`;

  const docComment: DocComment = parser.parseString(input);
  expect(docComment).toMatchSnapshot('docComment');
  expect(docComment.range.toString()).toMatchSnapshot('docComment.range');
});
