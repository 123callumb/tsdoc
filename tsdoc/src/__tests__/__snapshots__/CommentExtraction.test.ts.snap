// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment extraction: docComment 1`] = `
DocComment {
  "kind": 0,
  "lines": Array [],
  "parseErrors": Array [],
  "range": TextRange {
    "buffer": "
    /**
     * This is an example doc comment.
     * @remarks
     * Here are some remarks
     *
     * @public
     */
",
    "end": 8,
    "pos": 5,
  },
  "sourceRange": TextRange {
    "buffer": "
    /**
     * This is an example doc comment.
     * @remarks
     * Here are some remarks
     *
     * @public
     */
",
    "end": 123,
    "pos": 0,
  },
}
`;

exports[`Comment extraction: docComment.range 1`] = `"/**"`;
